Add GitLab integration documentation (#547)
Promptless/docs: src/content/docs/docs/integrations/gitlab-integration.mdx

---
title: GitLab Integration
slug: docs/integrations/gitlab-integration
sidebar:
  hidden: false
  order: 30
---
import Tip from '@components/fern/Tip.astro';
import Warning from '@components/fern/Warning.astro';

**Used for: Triggers**

> Promptless integrates with GitLab using personal access token authentication, monitoring merge requests in your repositories for documentation updates.

Reviewer: No, Promptless uses a Group Access Token.

<img width="512" height="327" alt="Image" src="https://github.com/user-attachments/assets/8c4763e7-9bc1-466c-8adc-01dde725ddcc" />

Reviewer: Here's the URL that's linked in the dialog: https://docs.gitlab.com/user/group/settings/group_access_tokens/

## Installation

To set up the GitLab integration:

> 1. Navigate to the **Integrations** page in your Promptless dashboard.

Reviewer: ```suggestion
1. Go to the [Integrations](https://app.gopromptless.ai/integrations) page in the Promptless dashboard.
```

> 2. Click on **Connect** next to the GitLab option.

Reviewer: ```suggestion
2. Click **Connect GitLab**.
```

3. Enter your GitLab credentials:
   - **Personal Access Token**: A GitLab personal access token with the required scope
   - **GitLab URL**: Your GitLab instance URL (use `https://gitlab.com` for GitLab.com)

<Warning title="Required Scope">

> Your GitLab personal access token must have the **`api`** scope. This scope provides full read/write access to the API, including all groups and projects, the container registry, and the package registry. Tokens without this scope will not work with the Promptless integration.

Reviewer: Use present tense and contractions when possible:

```suggestion
Your GitLab personal access token must have the **`api`** scope. Tokens without this scope don't work with Promptless.
```

---
title: GitLab Integration
slug: docs/integrations/gitlab-integration
sidebar:
  hidden: false
  order: 30
---
import Tip from '@components/fern/Tip.astro';
import Warning from '@components/fern/Warning.astro';

**Used for: Triggers**

Promptless integrates with GitLab using group access token authentication, monitoring merge requests in your repositories for documentation updates.

## Installation

To set up the GitLab integration:

1. Go to the [Integrations](https://app.gopromptless.ai/integrations) page in the Promptless dashboard.
2. Click **Connect GitLab**.
3. Enter your GitLab credentials:
   - **Group Access Token**: A GitLab group access token with the required scope

>    - **GitLab URL**: Your GitLab instance URL (use `https://gitlab.com` for GitLab.com)

Reviewer: ```suggestion
3. Enter a GitLab group access token with the `api` scope.
4. Click **Connect**.
```
